src/Language/PureScript/Scheme/CodeGen/Transpiler.hs

```diff
@@ -30,6 +36,13 @@ exprToScheme :: Expr Ann -> AST
 exprToScheme (Literal _ann literal) =
   literalToScheme literal
 
+exprToScheme (Constructor _ann _typeName constructorName fields) =
+  go fields
+  where
+    go (x:xs) = Lambda (runIdent x) (go xs)
+    go [] = cons (quote (Identifier (runProperName constructorName)))
+                 (vector (fmap (\field -> Identifier (runIdent field)) fields))
```

Comment:
Why do we turn a constructor into a lambda here? It would be good to have a comment explaining what's going on here